js/components/addressBook/addressBook_controller.js

```diff
@@ -4,8 +4,22 @@ angular.module('contactsApp')
 
 	ctrl.showUrl = false;
 	/* globals oc_config */
+
+	function compareVersion(version1, version2) {
+		var result=false;
+		for(var i=0;i<(Math.max(version1.length, version2.length));i++) {
+			if(Number(version1[i])<Number(version2[i])) {
+				result=true;
+				break;
+			}
+			if(version1[i]!==version2[i]) {
```

Comment:
Yeah, I snapped this code from somewhere and didn't even realised how poorly it was written...
Please forgive me! 😢